src/mscorlib/shared/System/Guid.cs

```diff
@@ -951,53 +986,68 @@ private static String EatAllWhitespace(String str)
             for (int i = 0; i < str.Length; i++)
             {
                 curChar = str[i];
-                if (!Char.IsWhiteSpace(curChar))
+                if (!char.IsWhiteSpace(curChar))
                 {
                     chArr[newLength++] = curChar;
                 }
             }
 
             // Return a new string based on chArr
-            return new String(chArr, 0, newLength);
+            return new string(chArr, 0, newLength);
         }
 
-        private static bool IsHexPrefix(String str, int i)
+        private static bool IsHexPrefix(string str, int i)
         {
-            if (str.Length > i + 1 && str[i] == '0' && (Char.ToLowerInvariant(str[i + 1]) == 'x'))
+            if (str.Length > i + 1 && str[i] == '0' && (char.ToLowerInvariant(str[i + 1]) == 'x'))
                 return true;
             else
                 return false;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public unsafe void WriteByteHelper(Span<byte> destination)
+        {
+            fixed (byte* d = &destination.DangerousGetPinnableReference())
```

Comment:
Why does this need to be pinned?  Does the perf measurably suffer if you just access destination[x] instead of d[x]?

Reply:
I was testing for perf differences and forgot to revert it- the perf doesn't measurably change. 
Changing to directly access destination[x].